tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge/work_order.cpp

```diff
@@ -17,15 +17,111 @@
 
 #include "tcf_error.h"
 #include "error.h"
-#include "avalon_sgx_error.h"
 #include "log.h"
 #include "types.h"
 
 #include "enclave.h"
 #include "base.h"
 #include "work_order.h"
 
-tcf_err_t WorkOrderHandlerBase::GetSerializedResponse(
+WorkOrderHandler::WorkOrderHandler(EnclaveType enclave_type) {
+    // Initialize function pointers
+    if (enclave_type == SINGLETON_ENCLAVE) {
```

Comment:
This is ok for time being. Later when we have multiple enclaves libraries for Singleton, KME and WPE then each enclave will have similar function names. So use Polymorphism and Virtual Functions instead of Function Pointers later